cmd/client-s3.go

```diff
@@ -507,6 +540,58 @@ func (c *s3Client) Copy(source string, size int64, progress io.Reader) *probe.Er
 	return nil
 }
 
+// PutObjectMetadata - upload an object with custom metadata.
+func (c *s3Client) PutWithMetadata(reader io.Reader, size int64, metadata map[string][]string, progress io.Reader) (int64, *probe.Error) {
+	bucket, object := c.url2BucketAndObject()
+	_, ok := metadata["Content-Type"]
```

Comment:
That is okay but client should set one of its own.  S3 doesn't say that if client should set or not set.   So a server implementation doesn't govern the client behavior.